superscore/client.py

```diff
@@ -205,59 +230,106 @@ def apply(
         else:
             return self.cl.put(pv_list, data_list)
 
-    def _gather_data(
+    def _gather_pvs(
         self,
-        entry: Union[Setpoint, Snapshot, UUID],
-        pv_list: Optional[List[str]] = None,
-        data_list: Optional[List[Any]] = None
-    ) -> Optional[tuple[List[str], List[Any]]]:
+        entry: Union[Collection, Parameter, UUID],
+    ) -> List[str]:
         """
-        Gather writable pv name - data pairs recursively.
-        If pv_list and data_list are provided, gathered data will be added to
-        these lists in-place. If both lists are omitted, this function will return
-        the two lists after gathering.
+        Collect all PV names accessible from ``entry``. Similar to _gather_data,
+        except this function does not associate data with the PVs.
 
-        Queries the backend to fill any UUID values found.
+        Parameters
+        ----------
+        entry : Union[Collection, Parameter, UUID]
+            Entry to gather PVs from
+
+        Returns
+        -------
+        List[str]
+            the list of PV names
+        """
+        pvs, _ = self._gather_data(entry, writable_only=False)
+        return pvs
+
+    def _gather_data(
+        self,
+        entry: Union[Entry, UUID],
+        writable_only: bool = False,
+    ) -> tuple[List[str], Optional[List[Any]]]:
+        """
+        Gather PV name - data pairs that are accessible from ``entry``.  Queries
+        the backend to fill any UUIDs found.
 
         Parameters
         ----------
-        entry : Union[Setpoint, Snapshot, UUID]
-            Entry to gather writable data from
-        pv_list : Optional[List[str]], optional
-            List of addresses to write data to, by default None
-        data_list : Optional[List[Any]], optional
-            List of data to write to addresses in ``pv_list``, by default None
+        entry : Union[Entry, UUID]
+            Entry to gather data from
+        writable_only : bool
+            If True, only include writable data e.g. omit Readbacks; by default False
 
         Returns
         -------
-        Optional[tuple[List[str], List[Any]]]
+        tuple[List[str], Optional[List[Any]]]
             the filled pv_list and data_list
         """
-        top_level = False
-        if (pv_list is None) and (data_list is None):
-            pv_list = []
+        pv_list = []
+        data_list = None
+        if isinstance(entry, Snapshot) or isinstance(entry, Setpoint) or isinstance(entry, Readback):
```

Comment:
```suggestion
        if isinstance(entry, (Snapshot, Setpoint, Readback)):
```

`isinstance` takes tuples as the second arg

Reply:
I thought I tried this and got an error, but I think I must have tried a `Union`, which doesn't work with python 3.9.  Thanks for the catch